### ISSUE TITLE
[Question] Training code

### ISSUE THREAD

**Issue by tyhuang98** [NIRVANALAN/STream3R#12]
Thanks for your interesting work! I have some question on the training code.
1. The `python fast3r/train.py experiment=stream3r/stream3r` in readme seems to be a mistake? 
2. How to set a multi-node training job with multiple GPU nodes?
3. Why the `allow_repeat` parameter is set to true in the configs/experiment/stream3r/stream3r.yaml?

Thank you again!

**Comment by tyhuang98** [comment]
I have another question: it seems that some functions are missed, e.g., ` from stream3r.dust3r.datasets.utils.transforms import convert_input_to_pred_format, vis_track` and ` from stream3r.viz.viser_visualizer_track import start_visualization` are both non-existent.

**Comment by Luo-Yihang** [comment]
Oh yes, those are some debugging codes we used to visualize the dataset. And they are used in training or inference. Thank you!

**Comment by Luo-Yihang** [comment]
Sure, but the code is a bit dirty. I might open-source it on my own fork for your reference these days. I’ll update here once it’s available.

**Comment by Luo-Yihang** [comment]
> To add test set, can I directly add the validation_datasets behind the train_datasets on `configs/experiment/stream3r/stream3r.yaml`? Any other adjustments need to be conducted?

Hi! You can add `validation_datasets` since the project uses the Lightning training framework. However, you may also want to define your own validation logic in two places:

1. **[multiview_dust3r_module.py (L204–L205)](https://github.com/NIRVANALAN/STream3R/blob/03d6dd0f464e52859b36d5ea0cf2f86f1c43c806/stream3r/models/multiview_dust3r_module.py#L204-L205)**  
   Here, you can specify the validation logic in the `validation_step` method. Currently, it just returns the loss.

2. **[multiview_dust3r_datamodule.py (L129–L130)](https://github.com/NIRVANALAN/STream3R/blob/03d6dd0f464e52859b36d5ea0cf2f86f1c43c806/stream3r/data/multiview_dust3r_datamodule.py#L129-L130)**  
   The validation dataloader is slightly different from the training dataloader.
